### PR TITLE
MessagePanelTag doesn't close with </div> when it's empty #554 

### PR DESCRIPTION
**Pull request by btkatoutmj**
Please review #554 .

### CODE REVIEW

**Comment by kazuki43zoo** [comment]
@btkatoutmj 

Please add test cases for this issue.

**Comment by btkatoutmj** [comment]
@kazuki43zoo 
I added a test case for this issue.

**Comment by ikeyat** [comment]
@btkatoutmj 
 You need to add test cases for pagination tag.

File: terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java [hunk 1]
```diff
@@ -618,6 +618,20 @@ public void test33() throws Exception {
...
+     * check that start tags are not self-closing in messagesPanelTag.
+     */
+    @Test
+    public void test34() throws Exception {
+        request.setAttribute(ResultMessages.DEFAULT_MESSAGES_ATTRIBUTE_NAME,
+                new String[] {});
+        int ret = tag.doStartTag();
+        String expected = "<div class=\"alert\"><ul></ul></div>";
+        assertThat(getOutput(), is(expected));
+        assertThat(ret, is(TagSupport.EVAL_BODY_INCLUDE));
```
Comment (ikeyat) [inline terasoluna-gfw-web-jsp/src/test/java/org/terasoluna/gfw/web/message/MessagesPanelTagTest.java]: - [ ] Maybe this test does not cover for ``<li></li>``.

**Comment by ikeyat** [comment]
@iwanagah @kazuki43zoo 
Please confirm. If nothing to be modified, please merge this.

**Comment by iwanagah** [comment]
I've confirmed this.  There is nothing to be modified. Please check @kazuki43zoo

**Comment by kazuki43zoo** [comment]
Hi @ikeyat @iwanagah @btyoshikawaa @btkatoutmj ,

I want to discuss about expected value when messages is empty list.
In current implementation, it write `<div class=\"alert\"><ul></ul></div>`. Is correct this result ?　I think there is the option that not write nothing as same with when messages is null.

What do you think ?

File: terasoluna-gfw-web-jsp/src/main/java/org/terasoluna/gfw/web/message/MessagesPanelTag.java [hunk 2]
```diff
@@ -196,6 +196,11 @@ TagWriter createTagWriter() {
 
     /**
      * Creates messagesPanel tag
+     * <p>
+     * If message is {@code null}, this method doesn't do anything.<br>
+     * Otherwise, this method write {@link #panelElement} and {@link #outerElement},<br>
+     * and call {@link #writeMessages(TagWriter, Object)}.
+     * </p>
```
Comment (iwanagah) [inline terasoluna-gfw-web-jsp/src/main/java/org/terasoluna/gfw/web/message/MessagesPanelTag.java]: Please add sample parameter and sample HTML to class Javadoc.

**Comment by ikeyat** [comment]
This commnet is for a reminder.

@kazuki43zoo 
As the result, we do not change spec of output for empty message in order to keep the structure of DOM.
Because the output may impact the looks on the browser, we clarify the spec in javadoc.

**Comment by ikeyat** [comment]
@iwanagah 
It looks OK for me. Please merge after your confirmation.

**Comment by iwanagah** [comment]
OK. I will merge this.

**Comment by iwanagah** [comment]
Please backport

**Pull request merged**